t/22-svirt.t

```diff
@@ -699,4 +699,16 @@ subtest 'get_wait_still_screen_on_here_doc_input' => sub {
     is($svirt->get_wait_still_screen_on_here_doc_input({}), 0, 'wait_still_screen on here doc is not set for kvm');
 };
 
+subtest 'Upload assets' => sub {
+    my $osutils_mock = Test::MockModule->new('osutils');
+    $osutils_mock->redefine(run => sub {
+            my @cmd = @_;
+            is 'nice ionice qemu-img convert -p -O qcow2 assets/barfoob.img assets/foo -c', join(' ', @cmd), 'got expected command';
+            return 0, '';
+    });
+
+    my $args = {name => 'foo', hdd_num => 1, format => 'qcow2', dir => 'assets'};
+    is($svirt->do_extract_assets($args), undef, 'Assets extracted');
```

Comment:
This is the code for qemu:
https://github.com/os-autoinst/os-autoinst/blob/master/t/14-isotovideo.t#L178
```
    combined_like {
        $return_code = handle_generated_assets($command_handler, 1)
    } qr/Requested to force the publication/, 'forced publication of asset';
    is $return_code, 0, 'The asset was uploaded successfully' or die path(bmwqemu::STATE_FILE)->slurp;
    my $force_publish_asset = $pool_dir . '/assets_public/force_publish_test.qcow2';
    ok(-e $force_publish_asset, 'test.qcow2 image exists');
    ok(!-e $pool_dir . '/assets_public/publish_test.qcow2', 'the asset defined by PUBLISH_HDD_X would not be generated in an incomplete job');
```

Reply:
Thanks! I changed my test now to also use `handle_generated_assets` which I think is better than directly calling into the backend